### PR TITLE
Use ConfigureAwait.Fody

### PR DESCRIPTION
**Pull request by Tyrrrz**
Imagine having to write `ConfigureAwait(false)` in 2020.

### CODE REVIEW

**Comment by Tyrrrz** [comment]
Hmm... CI is just stuck for some reason.

**Review by bruno-garcia — COMMENTED** [review]
If you wanted to net some lines here, it backfire: 😄 
image

What about we add the `xml` files on the `src` folder and create a symlink?

**Comment by Tyrrrz** [comment]
> If you wanted to net some lines here, it backfire:

True, but those added lines are configuration files, not in code ;)

> What about we add the xml files on the src folder and create a symlink?

Could try. I'm not sure how to do that.

**Comment by bruno-garcia** [comment]
> Could try. I'm not sure how to do that.

Try adding the xml and xsd file to `src`, then on each project: `ln -s ../bla.xml bla.xml` to create `bla.xml` locally as a symlink to the parent folder's file. This way we have 1 file, and all projects just link to it.

**Comment by Tyrrrz** [comment]
That's new

**Comment by Tyrrrz** [comment]
Interesting, the tests pass in debug mode, but freeze up in release mode.

**Comment by Tyrrrz** [comment]
Very interesting. Looks like in `Release` configuration the `.Do(p => ...)` delegate is not getting invoked at all, so as the result the MRE doesn't get triggered and the test hangs.

**Comment by bruno-garcia** [comment]
Maybe a regression on the mocking lib? This is odd

File: src/Sentry.AspNetCore/FodyWeavers.xsd [hunk 1]
```diff
@@ -0,0 +1,30 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (bruno-garcia) [inline src/Sentry.AspNetCore/FodyWeavers.xsd:1]: Anyv luck with the symlinks? These 300 lines of xml make me nervous

**Comment by Tyrrrz** [comment]
No idea why it worked before (c), but now I changed it to trigger the signal on a separate thread, which avoids the deadlock.

File: src/Sentry.AspNetCore/FodyWeavers.xsd (hunk 1 above)
Comment (Tyrrrz) [reply src/Sentry.AspNetCore/FodyWeavers.xsd:1]: Will look into this next

**Comment by Tyrrrz** [comment]
Lol, now `BackgroundWorkerTests` fail on release but not on debug. Interesting.

**Comment by bruno-garcia** [comment]
I'll blame it on IL magic breaking stuff :P

**Comment by Tyrrrz** [comment]
Damn you're right:

image

My guess is that NSubstitute and Fody both muddle with IL and it ends up in this weird state. I don't know what to do here in the short term, apart from switching to Moq in hopes that it helps.

File: src/Sentry.AspNetCore/FodyWeavers.xsd (hunk 1 above)
Comment (Tyrrrz) [reply src/Sentry.AspNetCore/FodyWeavers.xsd:1]: I seem to have created a symlink (using `mklink` on Windows tho), but not sure how git is supposed to handle that. Reading [here](https://stackoverflow.com/questions/954560/how-does-git-handle-symbolic-links), it seems that git still handles them as separate files (which seems to be confirmed by `git diff`):
image

**Comment by bruno-garcia** [comment]
> My guess is that NSubstitute and Fody both muddle with IL and it ends up in this weird state. I don't know what to do here in the short term, apart from switching to Moq in hopes that it helps.

I propose we just write `ConfigureAwait(false)` then. It's just "ugly", there's no downside in affecting customers usage, performance. size of the binary or anything other than make programmer writing the SDK write boiler plate.

File: src/Sentry.AspNetCore/FodyWeavers.xsd (hunk 1 above)
Comment (Tyrrrz) [reply src/Sentry.AspNetCore/FodyWeavers.xsd:1]: Nevermind, apparently I was creating a hardlink and not a symlink (no idea what the difference is, but it seems to work now).

**Comment by Tyrrrz** [comment]
> ...anything other than make programmer writing the SDK write boiler plate.

Which results in noise that increases cognitive load. But yeah, I agree, it seems that we reached a point where it conflicts with other dependencies.

**Comment by bruno-garcia** [comment]
> Which results in noise that increases cognitive load. But yeah, I agree, it seems that we reached a point where it conflicts with other dependencies.

Wanna contribute to the Objective-C SDK or Java 1.8 or some other goodie? :)

**Comment by Tyrrrz** [comment]
> > Which results in noise that increases cognitive load. But yeah, I agree, it seems that we reached a point where it conflicts with other dependencies.
> 
> Wanna contribute to the Objective-C SDK or Java 1.8 or some other goodie? :)

I can contribute my disdain!

**Pull request closed without merging**